xbmc/utils/HevcSei.cpp

```diff
@@ -0,0 +1,187 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "HevcSei.h"
+
+void hevc_add_start_code_emulation_prevention_3_byte(std::vector<uint8_t>& buf)
+{
+  size_t i = 0;
+
+  while (i < buf.size())
+  {
+    if (i > 2 && buf[i - 2] == 0 && buf[i - 1] == 0 && buf[i] <= 3)
+      buf.insert(buf.begin() + i, 3);
+
+    i += 1;
+  }
+}
+
+void hevc_clear_start_code_emulation_prevention_3_byte(const uint8_t* buf,
+                                                       const size_t len,
+                                                       std::vector<uint8_t>& out)
+{
+  size_t i = 0;
+
+  if (len > 2)
+  {
+    out.reserve(len);
+
+    out.push_back(buf[0]);
+    out.push_back(buf[1]);
+
+    for (i = 2; i < len; i++)
+    {
+      if (!(buf[i - 2] == 0 && buf[i - 1] == 0 && buf[i] == 3))
+        out.push_back(buf[i]);
+    }
+  }
+  else
+  {
+    out.assign(buf, buf + len);
+  }
+}
+
+int CHevcSei::ParseSeiMessage(CBitstreamReader& br, std::vector<CHevcSei>& messages)
+{
+  CHevcSei sei{};
```

Comment:
```suggestion
  CHevcSei sei;
```

As now class is self-initialized is not necessary here